src/main/kotlin/com/google/androidstudiopoet/DependencyValidator.kt

```diff
@@ -19,13 +19,64 @@ package com.google.androidstudiopoet
 import com.google.androidstudiopoet.models.DependencyConfig
 
 class DependencyValidator {
-    fun isValid(dependencies: List<DependencyConfig>, moduleCount: Int): Boolean {
+
+    private val ANDROID_TYPE = "androidAppModule"
+    private val MODULE_TYPE = "module"
+
+    fun isValid(dependencies: List<DependencyConfig>, moduleCount: Int, androidModuleCount: Int): Boolean {
+        for (dependency in dependencies) {
+            if (badDependency(dependency, moduleCount, androidModuleCount)) {
+                return false
+            }
+        }
         //TODO Add check for cycle dependencies and proper reporting
-        return correctAmountOfModules(dependencies, moduleCount)
+        return true
+    }
+
+    private class ModuleSplit(moduleName : String){
+        val type: String
+        val index: Int
+
+        init {
+            var firstChar = moduleName.length - 1
+            while ((firstChar >= 0) && moduleName[firstChar].isDigit()) {
+                firstChar--
+            }
+            type = moduleName.substring(0, firstChar + 1)
+            index = moduleName.substring(firstChar + 1).toInt()
+        }
+    }
+
+    private fun badDependency(dependency: DependencyConfig, moduleCount: Int, androidModuleCount: Int): Boolean {
+        val from = ModuleSplit(dependency.from)
+        // Check types and indexes
+        if (badIndexByType(from, moduleCount, androidModuleCount)) {
+            return true
+        }
+        val to = ModuleSplit(dependency.to)
+        if (badIndexByType(to, moduleCount, androidModuleCount)) {
+            return true
+        }
+        // check that a java module does not depend on an Android module
+        if (from.type.equals(MODULE_TYPE) && to.type.equals(ANDROID_TYPE)) {
```

Comment:
Please, instead of comment make a method with proper name. For the next `if` as well. Then it would be also possible to have one `when` instead of 5 `if`, but it is optional, if you like `if`s keep it like it is.

Reply:
Thanks for the comments! I have modified the code to address those issues and now the file looks cleaner! Please, take a look.